logging/Category.cpp

```diff
@@ -108,7 +105,11 @@ void Category::_logUnconditionally2(log4cpp::Priority::Value priority,
 
 void Category::callAppenders(const OCL::logging::LoggingEvent& event) throw()
 {
-    log_port.write( event );
+    // only write if the port is connected
+    if (log_port.connected())
```

Comment:
It should be noted that even if [ConnectionManager::connected()](https://github.com/orocos-toolchain/rtt/blob/f3660dbc86fd9e0681531c1e85596bc4e66275f6/rtt/internal/ConnectionManager.cpp#L127) at the moment does not lock the `connection_mutex`, this would actually be required for full thread-safety. In that case this additional check would not be effective.

I am not sure what is the best option to avoid unnecessary mutex locks and still allow thread-safe addition and deletion of connections while the component is active. We could add a compile- or run-time option to RTT to disable the mutexes in the `ConnectionManager` and in `ChannelElement`, disallowing destruction of connections while the writer is still active. I think a correct implementation of read/write-locks in [toolchain-2.9](https://github.com/orocos-toolchain/rtt/blob/toolchain-2.9/rtt/os/Mutex.hpp#L340) that is lock-free in case there are no writers would be the cleanest option. At the moment they are still using an internal mutex lock to protect the reader counter, like in [Boost Thread](https://github.com/boostorg/thread/blob/develop/include/boost/thread/pthread/shared_mutex.hpp#L188).

In toolchain-2.9 and with `keepsLastWrittenValue() == false` the check would be redundant as the OutputPort already checks connection status internally in [OutputPort.hpp:254](https://github.com/orocos-toolchain/rtt/blame/toolchain-2.9/rtt/OutputPort.hpp#L254).